rest/route/commit_queue.go

```diff
@@ -165,3 +168,75 @@ func (cq *commitQueueEnqueueItemHandler) Run(ctx context.Context) gimlet.Respond
 
 	return gimlet.NewJSONResponse(model.APICommitQueuePosition{Position: position})
 }
+
+type getCommitQueueItemAuthorHandler struct {
+	item      string
+	projectID string
+
+	sc data.Connector
+}
+
+func makeGetCommitQueueItemAuthor(sc data.Connector) gimlet.RouteHandler {
+	return &getCommitQueueItemAuthorHandler{
+		sc: sc,
+	}
+}
+
+func (cq *getCommitQueueItemAuthorHandler) Factory() gimlet.RouteHandler {
+	return &getCommitQueueItemAuthorHandler{
+		sc: cq.sc,
+	}
+}
+
+func (cq *getCommitQueueItemAuthorHandler) Parse(ctx context.Context, r *http.Request) error {
+	vars := gimlet.GetVars(r)
+	cq.item = vars["item"]
+	cq.projectID = gimlet.GetVars(r)["project_id"]
+	return nil
+}
+
+func (cq *getCommitQueueItemAuthorHandler) Run(ctx context.Context) gimlet.Responder {
+	pRef, err := cq.sc.FindProjectById(cq.projectID)
+	if err != nil {
+		return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "problem finding project"))
+	}
+
+	resp := model.APICommitQueueItemAuthor{}
+	if pRef.CommitQueue.PatchType == commitqueue.CLIPatchType {
+		p, err := cq.sc.FindPatchById(cq.item)
+		if err != nil {
+			return gimlet.NewJSONErrorResponse(errors.Wrap(err, "problem finding patch"))
+		}
+		resp.Author = model.ToAPIString(p.Author)
+		return gimlet.NewJSONResponse(resp)
+	}
+	if pRef.CommitQueue.PatchType == commitqueue.PRPatchType {
+		prNum, err := strconv.Atoi(cq.item)
+		if err != nil {
+			return gimlet.NewJSONErrorResponse(gimlet.ErrorResponse{
+				Message:    "must pass a valid PR number",
+				StatusCode: http.StatusBadRequest,
+			})
+		}
+		pr, err := cq.sc.GetGitHubPR(ctx, pRef.Owner, pRef.Repo, prNum)
+		if err != nil {
+			return gimlet.NewJSONErrorResponse(errors.Wrap(err, "problem finding pull request"))
+		}
+
+		var author string
+		if pr != nil && pr.User != nil && pr.User.ID != nil {
+			dbUser, _ := user.FindByGithubUID(*pr.User.ID)
+			if dbUser != nil {
+				author = dbUser.Username()
+			} else { // return github login if evergreen username unavailable
+				author = pr.User.GetLogin()
+			}
+		}
+		if author != "" {
+			resp.Author = model.ToAPIString(author)
+			return gimlet.NewJSONResponse(resp)
+		}
+		return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "unable to find author"))
```

Comment:
Doesn't this mean that author is never the empty string in https://github.com/evergreen-ci/evergreen/pull/2492/files#diff-7f9dabe9369339d415247be60a5d1b45R206?

Reply:
Hm no because we are ignoring the error, so the author will still be empty.